botx/models/entities.py

```diff
@@ -107,15 +124,21 @@ def check_that_type_matches_data(
             if mention_type in user_mention_types:
                 return mention_type
             raise ValueError(
```

Comment:
Let's refactor this function. All wrong cases should be at the beginning, raising value error. One successful case should be at the bottom.